### PR TITLE
Implement 'new' in C++.

### PR DESCRIPTION
**Pull request by BillyONeal**
This replaces the TypeScript implementation of new with a C++ one. Other changes:

* Now creates both vcpkg.json and vcpkg-configuration.json, which resolves "vcpkg new does not create vcpkg.json" https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1494967
* vcpkg-configuration.json can be embedded into vcpkg.json with --single-file.
* Now fills in the default registry with the current baseline git SHA.
* --name and --version are now mandatory since we are creating vcpkg.json, but they can be skipped with --application as per https://github.com/microsoft/vcpkg-tool/pull/605

image

image

### CODE REVIEW

File: src/vcpkg/commands.new.cpp [hunk 1]
```diff
@@ -1,15 +1,236 @@
...
+        "",
+        "Only one of --version-relaxed, --version-date, or --version-string may be specified.");
+    DECLARE_AND_REGISTER_MESSAGE(
+        NewSpecifyNameVersionOrApplication,
+        (),
+        "",
+        "Either specify --name and --version to produce a manifest intended for C++ libraries, or specify "
+        "--application to indicate that the manifest is not intended to be used as a port.");
+    DECLARE_AND_REGISTER_MESSAGE(NewNameCannotBeEmpty, (), "", "--name cannot be empty.");
+    DECLARE_AND_REGISTER_MESSAGE(NewVersionCannotBeEmpty, (), "", "--version cannot be empty.");
```
Comment (JavierMatosD) [inline src/vcpkg/commands.new.cpp:42]: Please place these messages in messages.h/cpp :)

Comment (JavierMatosD) [inline src/vcpkg/commands.new.cpp]: Please place these messages in messages.h/cpp :)

File: src/vcpkg/commands.new.cpp [hunk 2]
```diff
@@ -1,15 +1,217 @@
...
 
 namespace vcpkg::Commands
 {
+    ExpectedL<Json::Object> build_prototype_manifest(const std::string* name,
+                                                     const std::string* version,
+                                                     bool option_application,
+                                                     bool option_version_relaxed,
+                                                     bool option_version_date,
+                                                     bool option_version_string)
+    {
```
Comment (JavierMatosD) [inline src/vcpkg/commands.new.cpp:56]: This is just personal preference, but the function could be easier to digest if it were split into smaller chunks for each option and just pass the manifest as an argument.

**Review by JavierMatosD — APPROVED** [review]

File: src/vcpkg/commands.new.cpp (hunk 2 above)
Comment (BillyONeal) [reply src/vcpkg/commands.new.cpp:56]: I'm not sure I understand: the manifest is the return value? Can you show the declarations you'd be looking for?

**Review by vicroms — APPROVED** [review]

**Pull request merged**